File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOBleVoiceParamModel.kt
//
//  IDOBleVoiceParamModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///
/// Set phone volume for device event number

open class IDOBleVoiceParamModel(totalVolume: Int, currentVolume: Int) : IDOBaseModel {

    /// Total volume
    @SerializedName("total_voice")
    var totalVolume: Int = totalVolume

    /// Current volume
    @SerializedName("now_voice")
    var currentVolume: Int = currentVolume



    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }

}
    

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOBleVoiceParamModel.swift
//
//  IDOBleVoiceParamModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Set phone volume for device event number
@objcMembers
public class IDOBleVoiceParamModel: NSObject, IDOBaseModel {
    /// Total volume
    public var totalVolume: Int
    /// Current volume
    public var currentVolume: Int

    enum CodingKeys: String, CodingKey {
        case totalVolume = "total_voice"
        case currentVolume = "now_voice"
    }

    public init(totalVolume: Int, currentVolume: Int) {
        self.totalVolume = totalVolume
        self.currentVolume = currentVolume
    }

    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}
